src/game/client/draw/w3dtreedraw.cpp

```diff
@@ -0,0 +1,117 @@
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3dtreedraw.h"
+#include "baseheightmap.h"
+#include "drawable.h"
+
+W3DTreeDrawModuleData::W3DTreeDrawModuleData() :
+    m_moveOutwardTime(1),
+    m_moveInwardTime(1),
+    m_moveOutwardDistanceFactor(1.0f),
+    m_darkeningFactor(0.0f),
+    m_toppleFX(nullptr),
+    m_bounceFX(nullptr),
+    m_killWhenFinishedToppling(true),
+    m_doTopple(false),
+    m_doShadow(false),
+    m_initialVelocityPercent(0.2f),
+    m_initialAccelPercent(0.01f),
+    m_bounceVelocityPercent(0.3f),
+    m_minimumToppleSpeed(0.5f),
+    m_sinkTime(300),
+    m_sinkDistance(20.0f)
+{
+}
+
+void W3DTreeDrawModuleData::Build_Field_Parse(MultiIniFieldParse &p)
+{
+    static FieldParse dataFieldParse[] = {
+        { "ModelName", &INI::Parse_AsciiString, 0, offsetof(W3DTreeDrawModuleData, m_modelName) },
+        { "TextureName", &INI::Parse_AsciiString, 0, offsetof(W3DTreeDrawModuleData, m_textureName) },
+        { "MoveOutwardTime", &INI::Parse_Duration_Unsigned_Int, 0, offsetof(W3DTreeDrawModuleData, m_moveOutwardTime) },
+        { "MoveInwardTime", &INI::Parse_Duration_Unsigned_Int, 0, offsetof(W3DTreeDrawModuleData, m_moveInwardTime) },
+        { "MoveOutwardDistanceFactor", &INI::Parse_Real, 0, offsetof(W3DTreeDrawModuleData, m_moveOutwardDistanceFactor) },
+        { "DarkeningFactor", &INI::Parse_Real, 0, offsetof(W3DTreeDrawModuleData, m_darkeningFactor) },
+        { "ToppleFX", &INI::Parse_FX_List, 0, offsetof(W3DTreeDrawModuleData, m_toppleFX) },
+        { "BounceFX", &INI::Parse_FX_List, 0, offsetof(W3DTreeDrawModuleData, m_bounceFX) },
+        { "StumpName", &INI::Parse_AsciiString, 0, offsetof(W3DTreeDrawModuleData, m_stumpName) },
+        { "KillWhenFinishedToppling", &INI::Parse_Bool, 0, offsetof(W3DTreeDrawModuleData, m_killWhenFinishedToppling) },
+        { "DoTopple", &INI::Parse_Bool, 0, offsetof(W3DTreeDrawModuleData, m_doTopple) },
+        { "InitialVelocityPercent",
+            &INI::Parse_Percent_To_Real,
+            0,
+            offsetof(W3DTreeDrawModuleData, m_initialVelocityPercent) },
+        { "InitialAccelPercent", &INI::Parse_Percent_To_Real, 0, offsetof(W3DTreeDrawModuleData, m_initialAccelPercent) },
+        { "BounceVelocityPercent",
+            &INI::Parse_Percent_To_Real,
+            0,
+            offsetof(W3DTreeDrawModuleData, m_bounceVelocityPercent) },
+        { "MinimumToppleSpeed",
+            &INI::Parse_Positive_Non_Zero_Real,
+            0,
+            offsetof(W3DTreeDrawModuleData, m_minimumToppleSpeed) },
+        { "SinkDistance", &INI::Parse_Positive_Non_Zero_Real, 0, offsetof(W3DTreeDrawModuleData, m_sinkDistance) },
+        { "SinkTime", &INI::Parse_Duration_Unsigned_Int, 0, offsetof(W3DTreeDrawModuleData, m_sinkTime) },
+        { "DoShadow", &INI::Parse_Bool, 0, offsetof(W3DTreeDrawModuleData, m_doShadow) }
+    };

```

Comment:
afaik you also need to have a null entry at the end of the parse table so that the INI functions know when to stop looking.

Reply:
fixed this and a few others that were missing the nullptrs at the end.